File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mrqc/qcreport/d/QcReportGrpDO.java
package iih.ci.mrqc.qcreport.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.mrqc.qcreport.d.desc.QcReportGrpDODesc;
import java.math.BigDecimal;

/**
 * 质控报表分组 DO数据 
 * 
 */
public class QcReportGrpDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_QCREPORTGRP= "Id_qcreportgrp";
	public static final String ID_GRP= "Id_grp";
	public static final String ID_ORG= "Id_org";
	public static final String NAME= "Name";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_qcreportgrp() {
		return ((String) getAttrVal("Id_qcreportgrp"));
	}	
	public void setId_qcreportgrp(String Id_qcreportgrp) {
		setAttrVal("Id_qcreportgrp", Id_qcreportgrp);
	}
	public String getId_grp() {
		return ((String) getAttrVal("Id_grp"));
	}	
	public void setId_grp(String Id_grp) {
		setAttrVal("Id_grp", Id_grp);
	}
	public String getId_org() {
		return ((String) getAttrVal("Id_org"));
	}	
	public void setId_org(String Id_org) {
		setAttrVal("Id_org", Id_org);
	}
	public String getName() {
		return ((String) getAttrVal("Name"));
	}	
	public void setName(String Name) {
		setAttrVal("Name", Name);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_qcreportgrp";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_qc_report_grp";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(QcReportGrpDODesc.class);
	}
	
}

File: client/iih_stub/iih.ci_stub/iih.ci.mrqc/qcreport/d/QcReportGrpDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.mrqc.qcreport.d
{
    /// <summary>
    /// QcReportGrpDO 的摘要说明。
    /// </summary>
    public class QcReportGrpDO : BaseDO {

        public QcReportGrpDO() {
        }
		public string Id_qcreportgrp {
            get { return getAttrVal<string>("Id_qcreportgrp",null); }
            set { setAttrVal<string>("Id_qcreportgrp", value); }
        }
		public string Id_grp {
            get { return getAttrVal<string>("Id_grp",null); }
            set { setAttrVal<string>("Id_grp", value); }
        }
		public string Id_org {
            get { return getAttrVal<string>("Id_org",null); }
            set { setAttrVal<string>("Id_org", value); }
        }
		public string Name {
            get { return getAttrVal<string>("Name",null); }
            set { setAttrVal<string>("Name", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_qc_report_grp";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_qcreportgrp";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.mrqc.qcreport.d.QcReportGrpDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_qcreportgrp", "Id_grp", "Id_org", "Name"};
        }
        
    }
}
